### PR TITLE
fix: Permission issue

### LINKED ISSUE #2447: Can't navigate to Users folder in Open
## Is your feature request related to a problem? Please describe.

 I'm always frustrated when I am opening the bot:
I can't enter the path, I can't even find Users folder!
image

### PR DESCRIPTION
**Pull request by liweitian**
## Description
glob in globby failed to scan some directory because of a permission issue.
But not sure what to modify to make glob work. @xieofxie and I have the same permission on that directory. So I replace glob with node fs to check whether a folder is a bot folder.
## Task Item
Closes #2447 
## Screenshots

### CODE REVIEW

**Comment by liweitian** [comment]
@xieofxie Could you help to test whether this pr solves your problem?

**Comment by xieofxie** [comment]
> @xieofxie Could you help to test whether this pr solves your problem?

Solved. Thanks!

**Comment by liweitian** [comment]
@a-b-r-o-w-n could you have a look? It is tested by xieofxie.

File: Composer/packages/server/src/services/storage.ts [hunk 1]
```diff
@@ -126,9 +126,12 @@ class StorageService {
...
     // locate new structure bot:
-    const dialogFiles = await storage.glob('*.dialog', path);
-    const isNewBot = dialogFiles.length > 0;
-
+    const children = await storage.readDir(path);
+    const dialogfile = /.+(.)dialog/;
+    let isNewBot = false;
+    if (children.find(childName => dialogfile.test(childName))) {
+      isNewBot = true;
+    }
```
Comment (a-b-r-o-w-n) [inline Composer/packages/server/src/services/storage.ts:134]:
```suggestion
    const isNewBot = children.some(name => dialogFile.test(name))
```

File: Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx [hunk 2]
```diff
@@ -99,7 +99,10 @@ export const DefineConversation: React.FC<DefineConversationProps> = props => {
   };
 
   useEffect(() => {
-    const currentPath = Path.join(focusedStorageFolder.parent, focusedStorageFolder.name);
+    const currentPath =
+      Object.keys(focusedStorageFolder).length > 0
+        ? Path.join(focusedStorageFolder.parent, focusedStorageFolder.name)
+        : '';
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx:105]: I'm still confused about why this needs to happen. When do we fetch the data to populate `focusedStorageFolder`? To me that seems like an issue here. We should fetch that before rendering the file selector.

Comment (liweitian) [reply Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx:105]: This line is a hot fix. Currenly, we have Storage and FocusedStorageFolder in global state. In Storage, we saved a path which will be used to update FocusedStorageFolder. The code is like 
useEffect(()=> update FocusedStorageFolder, [Storage]). I will refactor this part soon to remove this useEffect

The reason here I have to add a null props checking is: 
If creating a new bot, it is like 
Click New bot Button => fetch storage => finish updating storage => fetch FocusedStorageFolder => finish updating FocusedStorageFolder => user click Next => show folder info.
If opening a bot
Click open bot button => fetch storage => finish updating storage=>  Need to show Folder info now but we do not get FocuesdStoragerFolder yet, the UI start to render => app crush

Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx:105]: Can you fix this now?

Like maybe show a loading spinner if focused storage folder hasn't been set yet?

Comment (cwhitten) [reply Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx:105]: Ping @liweitian. What is the plan here?

Comment (liweitian) [reply Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx]: We already have a spinner inside the fileSelector. I moved that part up to its parent component with checking on the  focusedStorageFolder.

**Pull request merged**